src/python/pants/backend/python/target_types.py

```diff
@@ -197,58 +221,85 @@ class PexAlwaysWriteCache(BoolField):
     value: bool
 
 
-class PexIgnoreErrors(BoolField):
+class PexIgnoreErrorsField(BoolField):
     """Should we ignore when PEX cannot resolve dependencies?"""
 
     alias = "ignore_errors"
     default = False
     value: bool
 
 
-class PexShebang(StringField):
+class PexShebangField(StringField):
     """For the generated PEX, use this shebang."""
 
     alias = "shebang"
 
 
-class PexEmitWarnings(BoolField):
+class PexEmitWarningsField(BoolField):
     """Whether or not to emit PEX warnings at runtime.
 
     The default is determined by the option `pex_runtime_warnings` in the `[python-binary]` scope.
     """
 
     alias = "emit_warnings"
 
-    def value_or_global_default(self, python_binary_defaults: PythonBinaryDefaults) -> bool:
+    def value_or_global_default(self, pex_binary_defaults: PexBinaryDefaults) -> bool:
         if self.value is None:
-            return python_binary_defaults.pex_emit_warnings
+            return pex_binary_defaults.emit_warnings
         return self.value
 
 
-class PythonBinary(Target):
+class PexBinary(Target):
     """A Python target that can be converted into an executable PEX file.
 
     PEX files are self-contained executable files that contain a complete Python environment capable
     of running the target. For more information, see https://www.pantsbuild.org/docs/pex-files.
     """
 
-    alias = "python_binary"
+    alias = "pex_binary"
     core_fields = (
         *COMMON_PYTHON_FIELDS,
         OutputPathField,
-        PythonBinarySources,
-        PythonBinaryDependencies,
-        PythonEntryPoint,
-        PythonPlatforms,
-        PexInheritPath,
-        PexZipSafe,
-        PexAlwaysWriteCache,
-        PexIgnoreErrors,
-        PexShebang,
-        PexEmitWarnings,
+        PexBinarySources,
+        PexBinaryDependencies,
+        PexEntryPointField,
+        PexPlatformsField,
+        PexInheritPathField,
+        PexZipSafeField,
+        PexAlwaysWriteCacheField,
+        PexIgnoreErrorsField,
+        PexShebangField,
+        PexEmitWarningsField,
     )
 
 
+class PythonBinary(PexBinary):
+    """A Python target that can be converted into an executable PEX file.
+
+    PEX files are self-contained executable files that contain a complete Python environment capable
+    of running the target. For more information, see https://www.pantsbuild.org/docs/pex-files.
+    """
+
+    alias = "python_binary"
+
+    def __init__(  # type: ignore[misc]  # This is `@final`, but it's fine to override here.
+        self,
+        unhydrated_values: Dict[str, Any],
+        *,
+        address: Address,
+        union_membership: Optional[UnionMembership] = None,
+    ) -> None:
+        warn_or_error(
+            removal_version="2.1.0.dev0",
+            deprecated_entity_description="the `python_binary` target",
+            hint=(
+                f"Use the target type `pex_binary`, rather than `python_binary` for {address}. "
+                "The behavior is identical."
```

Comment:
I originally wanted to add a `sed` snippet, but gave up. Sed has some differences between macOS and Linux that I personally find hard to navigate.

The snippet is easy with `rg` and `sd` (modern Rust alternative), but too few users have those installed.

Distributing a Python script felt super overkill.

--

Lmk if we should in fact add some snippet, or at least the regex `^python_binary\(`.

Reply:
We use `sed -Ee "..."` for mac and linux in out own scripts successfully and stably. Did you need som sed feature beyond `-E` extended regexps?:
```
$ git grep -w sed *.sh
build-support/bin/contributors.sh:    | sed -e 's/;.*$//'
build-support/bin/contributors.sh:    sed -E -e "s|^|+ |" >> CONTRIBUTORS.md
build-support/bin/release-changelog-helper.sh:        sed -Ee "s|^\(#([0-9]+)\)$|https://github.com/pantsbuild/pants/pull/\1|"
build-support/bin/release-changelog-helper.sh:        sed -Ee "s|^.*(https?://[^ ]+).*$|\1|" | \
build-support/bin/release-changelog-helper.sh:        sed -Ee "s|[/\.]+$||"
build-support/bin/release.sh:    PEX_VERSION="$(requirement pex | sed -e "s|pex==||")"
```

Reply:
`sed -Ee 's#^python_binary\(#pex_binary(#g'` would be nice. I wonder if #9434 would become more tolerable now since we already have libcst as a dependency? It would be able to act on BUILD files that aren't perfectly formatted is all (where `^python_binary()` may not necessarily catch all of them).

Reply:
A suggestion for *after* the 2.0 release is I believe how to appropriately consider #9434.

Reply:
Thank you John and Danny for the help!